src/components/SearchAndFilter.js

```diff
@@ -0,0 +1,167 @@
+import React from "react";
+import {Container, InputGroup, Button, Modal, Form, FormControl} from "react-bootstrap";
+import {BsSearch} from "react-icons/bs";
+//import {GrAddCircle} from "react-icons/gr";
+
+class SearchAndFilter extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      modalOpen: false,
+      name: "",
+      ingredients: [],
+      minTime: 0,
+      maxTime: 0,
+      createdBy: "",
+      inputFields: []
+    };
+  }
+
+  // Determines whether modal window is open or not
+  setModal = (flag) => {
+    this.setState({modalOpen: flag});
+  }
+
+  // Keeps track of the user input for recipe name
+  setName = (event) => {
+    this.setState({name: event.target.value})
+  }
+
+  // Handles getting the ingredients the user inputted to filter on
+  setIngredient = (index, event) => {
+    const ingredients = [...this.state.ingredients];
+    if (!ingredients[index]) {
+      ingredients.push(event.target.value);
+    } else {
+      ingredients[index] = event.target.value;
+    }
+    this.setState({ingredients: ingredients}, () => console.log(this.state.ingredients));
+  }
+
+  // Handles creating a new input field with an option to add more every time the user clicks the plus button
+  addInput = () => {
+    const {inputFields, ingredients} = this.state;
+    const inputs = [...inputFields];
+    const length = inputFields.length;
+    inputs.push(
+      <React.Fragment key={length}>
+        <label>Ingredient: </label>
+        <input type="text" value={ingredients[length]} onChange={(event) => this.setIngredient(length, event)}/>
+        <Button onClick={this.addInput} type="button">+</Button>
+        <br></br>
+      </React.Fragment>
+    )
+    this.setState({inputFields: inputs});
+  }
+
+  // Keeps track of the user input for min time
+  setMin = (event) => {
+    this.setState({minTime: event.target.value});
+  }
+
+  // Keeps track of the user input for max time
+  setMax = (event) => {
+    this.setState({maxTime: event.target.value});
+  }
+
+  // Keeps track of the user input for created by
+  setCreator = (event) => {
+    this.setState({createdBy: event.target.value});
+  }
+
+  // Returns a valid query parameter string depending on the ingredients state
+  createIngredientQueryParam = ()  => {
+    let res = "";
+    let {ingredients} = this.state;
+    for (let i in ingredients) {
+      res += `ingredients=${ingredients[i]}&`
+    }
+    return res;
+  }
+
+  // Filters what gets displayed based on the user inputs
+  filterRecipes = async () => {
+    fetch(`http:localhost:8080/recipes?page=${this.props.page}&limit=${this.props.limit}&
```

Comment:
your fetch shouldn't be "localhost:8080". This will be your actual server endpoint, whether it exists right now or now.

Also, the semantics is always http://...

Reply:
my server is on port 8080...so instead it should be the url for app on app engine?